tests/sdk/test_assets.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from substra.sdk import assets, exceptions
+
+
+@pytest.fixture
+def node_graph():
+    return {
+        key: list(range(key))
+        for key in range(10)
+    }
+
+
+def test_compute_ranks(node_graph):
+    visited = assets.compute_ranks(node_graph=node_graph)
+    for key, rank in visited.items():
+        assert key == rank
+
+
+def test_compute_ranks_failure(node_graph):
+    node_graph[5].append(9)
+    with pytest.raises(exceptions.InvalidRequest):
```

Comment:
Could we have a more specific error here? Something dedicated to circular dependencies maybe.

Reply:
If we want the tests on substra-tests to pass, the SDK must return an InvalidRequest error but we can also add a check on the message content

Extract from the test in substra-tests (substra-tests/tests/test_execution_compute_plan.py::test_compute_plan_circular_dependency_failure):
```
with pytest.raises(substra.exceptions.InvalidRequest) as e:
        client.add_compute_plan(cp_spec)

    assert 'missing dependency among inModels IDs' in str(e.value)
```

Reply:
Ok, let's leave it like this then, thanks.